app/controllers/spina/journal/primer_theme/articles_controller.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+module Spina
+  module Journal
+    module PrimerTheme
+      # User-facing controller for journal articles
+      class ArticlesController < ApplicationController
+        before_action :set_article, :set_issue, :set_breadcrumb, :set_metadata
+
+        def show
+          add_breadcrumb @article.title
+        end
+
+        private
+
+        def set_article
+          @article = Admin::Journal::Article.includes(affiliations: [:institution]).find(params[:id])
+          @article.view_context = view_context
+        rescue ActiveRecord::RecordNotFound
+          send_file Rails.root.join('public/404.html'), type: 'text/html; charset=utf-8', status: 404
```

Comment:
We should probably find a better way of doing this ultimately (like a protected method in a top-level application controller to handle the exception)

Reply:
100%